### PR TITLE
lib-user: Add ContentBox component

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Describe your changes
- add ContentBox component largely based on [app-project/src/shared/components/ContentBox](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/app-project/src/shared/components/ContentBox)
- related [Top Projects Figma design](https://www.figma.com/file/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?type=design&node-id=402-10023&mode=dev)
- add ^ related tests and stories
- refactor ProjectCard to support `small` boolean property per [Top Projects split version](https://www.figma.com/file/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?type=design&node-id=386-8410&mode=dev) required small project cards

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - review storybooks linked as follows...
- Which storybook stories should be reviewed?
  - http://localhost:6008/?path=/story/components-shared-contentbox--top-projects-full 
  - note http://localhost:6008/?path=/story/components-shared-contentbox--top-projects-split is not for small/mobile screens; in small/mobile width the ContentBox child that wraps project cards will not be a grid
- Are there plans for follow up PR’s to further fix this bug or develop this feature? I'll revisit the carousel design aspect in a subsequent PR

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [ ] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-user/src/components/shared/ContentBox/components/ContentHeading.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { SpacedText } from '@zooniverse/react-components'
+import { Heading } from 'grommet'
+import { node } from 'prop-types'
+import styled from 'styled-components'
+
+const StyledHeading = styled(Heading)`
+  font-size: 16px;
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ContentBox/components/ContentHeading.js:7]:
```suggestion
  font-size: 1rem;
```

If someone increases the font size in their browser, we want this to increase too. Actually, you can use `rem` units in the `size` prop on Heading components if that's easier to read such as `size='1rem'`.

File: packages/lib-user/src/components/shared/ContentBox/components/ContentHeading.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  line-height: normal;
+`
+
+function ContentHeading ({
+  children
+}) {
+  return (
+    <StyledHeading
+      level={2}
+      margin='none'
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ContentBox/components/ContentHeading.js:17]:
```suggestion
      margin='0'
```

`none` works for Grommet, but it's not an actual option for the CSS [margin](https://developer.mozilla.org/en-US/docs/Web/CSS/margin#syntax) property. I don't think `none` will throw a warning to the console, but in places where we've used `margin='none'` or `pad='none'` such as [ZooniverseTalk](https://github.com/zooniverse/front-end-monorepo/blob/c2ae9c51850ebfbf95ee1f272e09726559424fff/packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/ZooniverseTalk.js#L33) in app-project, there's a warning in the dev tool's styling inspector:
Screenshot 2024-01-25 at 3 27 35 PM

File: packages/lib-user/src/components/shared/ContentBox/components/ContentLink.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { SpacedText } from '@zooniverse/react-components'
+import { Anchor } from 'grommet'
+import { shape, string } from 'prop-types'
+import styled from 'styled-components'
+
+const StyledAnchor = styled(Anchor)`
+  font-size: 16px;
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ContentBox/components/ContentLink.js:7]:
```suggestion
  font-size: 1rem;
```

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js [hunk 4]
```diff
@@ -1,37 +1,39 @@
...
-import { string } from 'prop-types'
-import styled from 'styled-components'
+import { bool, string } from 'prop-types'
+import styled, { css } from 'styled-components'
 import { SpacedText } from '@zooniverse/react-components'
 
 const StyledProjectCard = styled(Box)`
-  font-size: 11px;
-  height: 280px;
+  font-size: 10px;
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js:7]:
```suggestion
  font-size: 0.625rem;
```

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js [hunk 5]
```diff
@@ -1,37 +1,39 @@
...
 
-  @media (width <= 768px) {
-    font-size: 10px;
-    height: 200px;
-    width: 157px;
-  }
-`
+  ${props => !props.small && css`
+    @media (768px < width <= 1280px) {
+      font-size: 10.5px;
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js:18]:
```suggestion
      font-size: 0.656rem;
```

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js [hunk 6]
```diff
@@ -1,37 +1,39 @@
...
+      > .project-image {
+        height: 189px;
+      }
+    }
 
-  @media (768px < width < 1280px) {
-    height: 189px;
-  }
+    @media (width > 1280px) {
+      font-size: 11px;
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js:28]:
```suggestion
      font-size: 0.688rem;
```

**Review by goplayoutside3 — APPROVED** [review]
Looking good! Just a couple minor styling suggestions.

File: packages/lib-user/src/components/shared/ContentBox/components/ContentHeading.js (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-user/src/components/shared/ContentBox/components/ContentHeading.js]: Ah yes, just read that in the related nice article, thanks for flagging!

**Pull request merged**